src/overlap_check/ProgressBar.hpp

```diff
@@ -0,0 +1,24 @@
+
+#ifndef DAGMC_PROGRESSBAR_H
+#define DAGMC_PROGRESSBAR_H
+
+#include <string>
+
+class ProgressBar {
+
+ public:
+  // constructor
+  ProgressBar();
+
+  // destructor
+  ~ProgressBar();
+
+  void set_value(double val);
+
+  static bool is_terminal();
+
+ private:
+  int current {0};
+};
```

Comment:
Suggest addition of `bool need_final_newline {false};`. See ProgressBar.cpp for more details. This is to remove superfluous newline on successful completion [part 1 of 4].